lua/oil/init.lua

```diff
@@ -384,6 +384,12 @@ M.open = function(dir)
   if config.buf_options.buflisted ~= nil then
     vim.api.nvim_buf_set_option(0, "buflisted", config.buf_options.buflisted)
   end
+
+  -- If preview window exists, update its content
+  local preview_win_id = util.get_preview_win()
+  if preview_win_id then
+    M.select({ preview = true })
+  end
```

Comment:
One question I'm not sure about is: what to do when `update_on_cursor_moved = false`, specifically when moving into a folder/parent. Some ideas but no strong feelings yet:
1. Update the preview window (what I implemented)
2. Keep preview window open but don't update it
3. Close preview window completely

Reply:
I think either 1 or 2 would be fine. If you use this you'll need to make use of the `run_after_load` utility, though. Right now if you open the preview window and go to the parent directory, it displays a "Could not find entry under cursor" error.